mysql/datadog_checks/mysql/statements.py

```diff
@@ -0,0 +1,126 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+from contextlib import closing
+
+import pymysql
+
+from datadog_checks.base import AgentCheck, is_affirmative
+from datadog_checks.base.utils.db.sql import compute_sql_signature
+from datadog_checks.base.utils.db.statement_metrics import StatementMetrics
+
+try:
+    import datadog_agent
+except ImportError:
+    from ..stubs import datadog_agent
+
+
+METRICS = {
+    'count': ('mysql.queries.count', AgentCheck.count),
+    'errors': ('mysql.queries.errors', AgentCheck.count),
+    'time': ('mysql.queries.time', AgentCheck.count),
+    'select_scan': ('mysql.queries.select_scan', AgentCheck.count),
+    'select_full_join': ('mysql.queries.select_full_join', AgentCheck.count),
+    'no_index_used': ('mysql.queries.no_index_used', AgentCheck.count),
+    'no_good_index_used': ('mysql.queries.no_good_index_used', AgentCheck.count),
+    'lock_time': ('mysql.queries.lock_time', AgentCheck.count),
+    'rows_affected': ('mysql.queries.rows_affected', AgentCheck.count),
+    'rows_sent': ('mysql.queries.rows_sent', AgentCheck.count),
+    'rows_examined': ('mysql.queries.rows_examined', AgentCheck.count),
+}
+
+
+DEFAULT_METRIC_LIMITS = {
```

Comment:
This API is a bit difficult to reason about. 

* there's isn't an easy way to limit (or reason about) what the total cardinality is going to be.  It looks like the worst case cardinality is the sum of all the topk and bottomk numbers? It would be good to be able to set an absolute limit on the query cardinality.
* if the numbers are set too low so that we're missing some queries but we would be fine collecting more, then there isn't an easy way to say "fill up the rest with queries sorted by`column`"

Reply:
Right, the worst case would be sum of all limits. Note there is a hardcoded limit of 10000 in the query itself, which is the MySQL default size of that table. I can take another pass at filling or truncating metrics on a hard limit. Do you think it's important enough to block the existing changes?

Reply:
> I can take another pass at filling or truncating metrics on a hard limit. Do you think it's important enough to block the existing changes? 

Ok, yeah we can proceed and iterate later.